Moonscraper Chart Editor/Assets/Scripts/Game/Charts/IO/Midi/MidReader.cs

```diff
@@ -1171,26 +1208,23 @@ static void ProcessEventAsForcedTypePostDelay(in EventProcessParams eventProcess
             }
         }
 
-        static uint CalculateSustainLength(Song song, NoteOnEvent noteEvent)
+        static uint AdjustSustainLength(Song song, uint length)
```

Comment:
Went with a much more exact name, `CutoffSustainIfNeeded`

Reply:
This is probably gone too far the other way where it's become too on the nose for how it's being implemented. Imo a good function name should be performing one single task or goal. "CutoffSustainIfNeeded" exposes it's branching nature which could change later, and just raises questions like "why is a cutoff needed or not needed?" to someone reading it. Having "and" or "if" or similar words is something you should avoid with naming.

Reply:
Would something like `ApplySustainCutoff` be better? I don't like `Sanitise` because it's not exactly *sanitizing* anything. It's not like lengths beyond a certain bound can cause bad things lol, it's just applying a cutoff below a certain threshold.

Reply:
Sanitising in this case is just correcting for any errors or issues in the input and returning a potentially corrected version, which in this case our aim is to ensure that sustains aren't doing anything funny, specifically ensuring that sustains aren't being hidden below the note sprite/model. A "sustain cutoff" kinda implies that all sustains are going to get cutoff  or capped by some max length as I read it, which is the opposite of what it's actually doing. "Sanitise" is just a nice catch-all term for making sure that our sustain length values are sane/within reason. 
What it's really doing is ensuring a minimum sustain length, but if the function was called that then it wouldn't be unreasonable to assume that a sustain length of 0 should be raised to a sustain length of 1/12th based on the name alone. There's not really a concise way to explain this behaviour since it's really quite specific so it feels reasonable to use a catch-all error correction term for the function name and leave the specific correction behaviour to the function implementation. The caller doesn't need to know how the data it's being corrected, they just need to know that it may need to be corrected.

Reply:
Alright, whatever lol. I only changed the name in the first place since it's no longer really *calculating* a sustain length from a MIDI note, instead it's taking a pre-retrieved length and applying the sustain cutoff. If there was more that needed to be done to the length then I wouldn't be so opposed to 'sanitize' being the name, but that's literally all it does and most likely all it ever will do.

Also, the sustain cutoff isn't a visual thing, it's because without it every note in charts made with a MIDI editor would be a sustain when they're not supposed to be, as manually making every non-sustain note 1 tick is much too tedious.

Reply:
Yeah I agree it was a bad name to begin with but if we are going to be touching lines I'd rather change them to something that makes sense for other people reading the code now that it's open source.